tests/ArrayHelper/KeyAndPathExistsTest.php

```diff
@@ -0,0 +1,118 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Yiisoft\Arrays\Tests\ArrayHelper;
+
+use PHPUnit\Framework\TestCase;
+use Yiisoft\Arrays\ArrayHelper;
+
+final class KeyAndPathExistsTest extends TestCase
+{
+    private array $array = [
+        'name' => 'test',
+        'date' => '31-12-2113',
+        'post' => [
+            'id' => 5,
+            'author' => [
+                'name' => 'cebe',
+                'profile' => [
+                    'title' => '1337',
+                ],
+            ],
+        ],
+        'admin.firstname' => 'Qiang',
+        'admin.lastname' => 'Xue',
+        'admin' => [
+            'lastname' => 'cebe',
+        ],
+        'version' => [
+            '1.0' => [
+                'status' => 'released',
+                'name' => 'world',
+            ],
+            '1.0.status' => 'dev',
+            2 => 'two',
+        ],
+        '42.7' => 500,
+    ];
+
+    /**
+     * @return array[] common test data for [[testKeyExists()]] and [[testPathExists()]]
+     */
+    private function commonData(): array
```

Comment:
Let's remove this method and copy-paste these cases to needy methods

Reply:
This methods should be work different only for strings with dots. In all other cases this methods equally, so I create separately method with common data.

Also made for tests of getValue and getValueByPath methods.